compiler/src/iree/compiler/Codegen/LLVMGPU/TransformExtensions/LLVMGPUExtensions.cpp

```diff
@@ -1496,5 +1500,41 @@ void transform_dialect::PackSharedMemoryAllocOp::getEffects(
   transform::modifiesPayload(effects);
 }
 
+class TestVectorLayoutOptions : public VectorLayoutOptions {
+public:
+  TestVectorLayoutOptions(Operation *root) : VectorLayoutOptions(root) {}
+
+  void setAnchorOps(VectorLayoutAnalysis &analysis) override {
+    setAnchorOpsFromAttributes(analysis, root);
+  }
+};
+
+DiagnosedSilenceableFailure
+transform_dialect::TestAMDGPUContractionDistribution::applyToOne(
+    transform::TransformRewriter &rewriter, func::FuncOp target,
+    transform::ApplyToEachResultList &results,
+    transform::TransformState &state) {
+  TestVectorLayoutOptions options(target);
+  RewritePatternSet patterns(target.getContext());
+  int64_t m = getM();
+  int64_t n = getN();
+  int64_t k = getK();
+  MFMAType mfmaType;
+  if ((m == 16) && (n == 16) && (k == 16)) {
+    mfmaType = MFMAType::F16_16x16x16_F32;
```

Comment:
Is it not possible to determine the MFMA type from the layout? This looks like something the layout should tell us.

Reply:
For the 3 different matrix cores we have seen so far, I would agree. But in general, you can imagine a scenario where different instructions use the same layout. It seems like having the ability to infer a set of compatible instructions for a given layout does makes sense though, acting kind of like a verifier. I can modify this to reflect that.